lib/diagnostics_channel.js

```diff
@@ -0,0 +1,292 @@
+'use strict';
+
+const {
+  ArrayPrototypeIncludes,
+  ArrayPrototypePush,
+  ArrayPrototypeSplice,
+  ObjectCreate,
+  ObjectSetPrototypeOf,
+  Promise,
+  PromiseResolve,
+  RegExp,
+  Symbol
+} = primordials;
+
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+  }
+} = require('internal/errors');
+
+const { triggerUncaughtException } = internalBinding('errors');
+
+const { WeakReference } = internalBinding('util');
+
+function alwaysMatch() {
+  return true;
+}
+
+function strictEqualMatch(string) {
+  return function(value) {
+    return value === string;
+  };
+}
+
+function regexpMatch(regexp) {
+  return function(value) {
+    return regexp.test(value);
```

Comment:
```suggestion
    return RegExpPrototypeTest(regexp, value);
```

Might also make more sense to use `FunctionPrototypeBind` here at least.